src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/Common/GuestConfigurationCmdletBase.cs

```diff
@@ -0,0 +1,376 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.GuestConfiguration.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using Newtonsoft.Json.Linq;
+
+    using Microsoft.Azure.Commands.Common.Authentication;
+    using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+    using Microsoft.Azure.Commands.ResourceManager.Common;
+    using Microsoft.Azure.Commands.GuestConfiguration.Helpers;
+    using Microsoft.Azure.Commands.GuestConfiguration.Models;
+    using Microsoft.Azure.Management.GuestConfiguration;
+    using Microsoft.Azure.Management.ResourceManager;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.ErrorResponses;
+    using Microsoft.Azure.Management.ResourceManager.Models;
+
+    /// <summary>
+    /// Base class for Azure Guest configuration cmdlets
+    /// </summary>
+    public abstract class GuestConfigurationCmdletBase : AzureRMCmdlet
+    {
+        /// <summary>
+        /// Guest configuration client
+        /// </summary>
+        private IGuestConfigurationClient _guestConfigurationClient;
+
+        /// <summary>
+        /// Azure policy client
+        /// </summary>
+        private IPolicyClient _policyClient;
+
+        private string _initiativeId = null;
+
+        /// <summary>
+        /// Gets or sets the guest configuration client
+        /// </summary>
+        public IGuestConfigurationClient GuestConfigurationClient
+        {
+            get
+            {
+                return _guestConfigurationClient ?? 
+                    (_guestConfigurationClient = AzureSession.Instance.ClientFactory.CreateArmClient<GuestConfigurationClient>(DefaultContext, AzureEnvironment.Endpoint.ResourceManager));
+            }
+            set
+            {
+                _guestConfigurationClient = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the Policy client
+        /// </summary>
+        public IPolicyClient PolicyClient
+        {
+            get
+            {
+                return _policyClient ??
+                    (_policyClient = AzureSession.Instance.ClientFactory.CreateArmClient<PolicyClient>(DefaultContext, AzureEnvironment.Endpoint.ResourceManager));
```

Comment:
If you are goign to use the polciy client, you shoudl use the commonone, as suggested above.